src/utils/daos_hdlr.c

```diff
@@ -1668,57 +1673,44 @@ dm_disconnect(struct cmd_args_s *ap,
 	if (src_file_dfs->type == DAOS) {
 		if (is_posix_copy) {
 			rc = dfs_sys_umount(src_file_dfs->dfs_sys);
-			if (rc != 0) {
-				DH_PERROR_DER(ap, daos_errno2der(rc), "failed to unmount source");
-				rc = dfs_sys_umount(src_file_dfs->dfs_sys);
-				if (rc != 0)
-					DH_PERROR_DER(ap, daos_errno2der(rc),
-						      "failed to unmount source on retry");
```

Comment:
So these retries weren't needed after all? I think it's an improvement to remove them if we can, but I don't fully understand why they were here to begin with.

Reply:
This PR removes the need for them.

Rather than always checking for memory leaks the test in question now disables the memory leak test if the process outputs "DER_BUSY".  It makes the check less robust but does mean that the retries in the code aren't needed - it's the desire to not do this that I created the PR for initially.